graph/src/components/store.rs

```diff
@@ -198,15 +214,33 @@ pub struct StoreEvent {
     pub changes: Vec<EntityChange>,
 }
 
-/// A boxed `StoreEventStream`
-pub type StoreEventStreamBox = Box<Stream<Item = StoreEvent, Error = ()> + Send>;
+impl StoreEvent {
+    /// Extend `ev1` with `ev2`. If `ev1` is `None`, just set it to `ev2`
+    fn accumulate(ev1: &mut Option<StoreEvent>, ev2: StoreEvent) {
+        if let Some(e) = ev1 {
+            e.extend(ev2);
+        } else {
+            ev1.replace(ev2);
+        }
+    }
+
+    fn extend(&mut self, other: StoreEvent) {
+        if other.source > self.source {
```

Comment:
Initially I wanted to use `source` to convey to subscribers "we've made it this far in the chain" so they could detect things like block reversion. But as you said, since it's not used right now, I'll take it out.

Reply:
The one thing for which source is useful is tracing events as they go through the system in logs. I added a comment to the declaration of `StoreEvent` to that effect.